sway-lsp/src/core/session.rs

```diff
@@ -158,17 +158,29 @@ impl Session {
         let locked = false;
         let offline = false;
 
+        let profile = pkg::BuildProfile {
+            silent: silent_mode,
+            ..pkg::BuildProfile::debug()
+        };
+
         // TODO: match on any errors and report them back to the user in a future PR
         if let Ok(manifest) = pkg::ManifestFile::from_dir(&manifest_dir) {
             if let Ok(plan) = pkg::BuildPlan::from_lock_and_manifest(&manifest, locked, offline) {
+                // First, parse items to collect keyword tokens
+                if let Ok(source) = manifest.entry_string() {
+                    if let Ok(sway_build_config) =
+                        pkg::sway_build_config(manifest.dir(), &manifest.entry_path(), &profile)
+                    {
+                        parse_items::parse(source, Some(&sway_build_config), &self.token_map);
+                    }
+                }
```

Comment:
Is there no way we can collect these `storage` tokens during our regular traversal of items, rather than needing a whole distinct parse step? We should avoid parsing or type-checking more than we need to, especially considering how frequently LSP will do both.

Reply:
I agree we should limit the amount of parsing we need to do. One option is to also return the `sway_ast::Module` along with the `ParseProgram` and `TypedProgram` when calling `forc::check`. Would probably need to rework a bit of API in `sway-core` to make this ergonomic which might not be worth the trouble. 

The other option is to pass through the attributes into the `ParseProgram` and access the `storage` token from there. I was planning on doing this so we could access the `doc` attribute. Maybe it's worth just closing this PR and adding this functionality back in once that stage has been completed.